src/shared-library.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2019 Jamie Madill
+ *
+ * This file is part of the glmark2 OpenGL (ES) 2.0 benchmark.
+ *
+ * glmark2 is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * glmark2 is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * glmark2.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *  Jamie Madill
+ */
+#include "shared-library.h"
+#include <dlfcn.h>
+#include <stdio.h>
+
+namespace
+{
+// Try both system and development  extensions on posix systems. Prioritize the
+// development extension for users who are doing driver development.
+const char *extensions[] = { ".so", ".so.1" };
+}
+
+/******************
+ * Public methods *
+ ******************/
+SharedLibrary::SharedLibrary() : handle_(0) {}
```

Comment:
Could use nullptr instead of 0 (most of the code doesn't use this due to being written pre-C++11).

Reply:
Done